contracts/interfaces/IRelayHub.sol

```diff
@@ -73,12 +74,16 @@ interface IRelayHub {
     /// @param OK - the transaction was successfully relayed and execution successful - never included in the event
     /// @param RelayedCallFailed - the transaction was relayed, but the relayed call failed
     /// @param PreRelayedFailed - the transaction was not relayed due to preRelatedCall reverting
+    /// @param ForwarderFailed - the transaction was not relayed due to forwrader check (signature,nonce)
+    /// @param TrustedRecipientFailed - the transaction was not relayed since Paymaster refuse to relay reverted recipient TX.
     /// @param PostRelayedFailed - the transaction was relayed and reverted due to postRelatedCall reverting
     /// @param RecipientBalanceChanged - the transaction was relayed and reverted due to the recipient balance change
     enum RelayCallStatus {
         OK,
         RelayedCallFailed,
         PreRelayedFailed,
+        ForwarderFailed,
+        RecipientFailed,
```

Comment:
What is the difference between `RelayedCallFailed` and `RecipientFailed`?

Reply:
there is, and I wonder if we should keep it:
- RecipientFailed is returned on `revertOnRecipientRevets` (gas calculation is done outside, but the "intent" is inside innerRelayedCall)
- RecipientCallFailed is returned if the paymaster want to pay anyway (that is, both pre and post are executed and not reverted)

I guest that even if we leave them both in, they need better naming..

Reply:
OMG, this is extremely confusing. Like, this flag is exposed upwards via a new value for the enum. Can we find a better solution?